datacompy/core.py

```diff
@@ -108,16 +109,18 @@ def __init__(
         self.df2_name = df2_name
         self.abs_tol = abs_tol
         self.rel_tol = rel_tol
-        self.df1_unq_rows = self.df2_unq_rows = self.intersect_rows = None
-        self.column_stats = []
+        self.df1_unq_rows: pd.DataFrame
```

Comment:
I think this and the next 2 need to be fixed also?

```
Optional[Type["pd.DataFrame"]] = None
```

Reply:
So those are now only the python 3.5 tests that are failing due to how I used type hints here (this was added in 3.6)

Reply:
OK, but I think we still want to support Python 3.5 for now. so we need to figure out a way to get the tests to pass given the type hinting changes. Anyone running on Python 3.5 will hit issues I'm guessing at runtime.

Reply:
Yeah, I'll fix it. It will be some effort and will probably involve me just ignoring a bunch of rows for mypy.

Reply:
The main issue is that mypy doesn't recognize the logic that set the variable (from `None` to a dataframe) and then complains when a dataframe method is used (because `None` of course doesn't have those).

Reply:
So all the methods that use, for example, `self.intersect_rows` fail mypy because it could be `None`. The solution to this that I used works only starting with 3.6. Other solutions would be to check in every method if the variable is `None` and in that case error out (which would technically prevent those bugs that mypy finds from happening) or ignore the types for all those rows (29 in total).

Reply:
Any preference?

Reply:
I opted for raising `TypeError` in case any of the attributes are `None`.